hist/hist/src/TFormula.cxx

```diff
@@ -998,50 +1000,113 @@ void TFormula::FillVecFunctionsShurtCuts() {
 ///    eg.
 ///    varpol2(5) will be replaced with: [5] + [6]*var + [7]*var^2
 ///    Empty name is treated like variable x.
+///    Extended format also supports direct variable specification: pol2(x, 0) or pol(x, [A])
+///    Traditional Syntax: polN - Polynomial ofreturn it->second;(x)
+///    ypol1             [p0]+[p1]*y
+///    pol1(x, 0)        [p0]+[p1]*x
+///    pol2(y, 1)        [p1]+[p2]*y+[p3]*TMath::Sq(y)
+///
+///    Extended Code Documentation
+///    The polynomial handling has been enhanced to support explicit variable names
+///    and parameter placeholders in polynomial expressions. This makes the syntax
+///    more consistent with other function types in ROOT (like gaus) and provides
+///    more flexibility in polynomial construction.
+///    The function reports errors through ROOT's error handling system
+///
+///    This documentation update reflects changes made to support
+///    GitHub issue #16794 for enhanced polynomial syntax support.
 
 void TFormula::HandlePolN(TString &formula)
 {
    Int_t polPos = formula.Index("pol");
    while (polPos != kNPOS && !IsAParameterName(formula, polPos)) {
-
       Bool_t defaultVariable = false;
       TString variable;
       Int_t openingBracketPos = formula.Index('(', polPos);
       Bool_t defaultCounter = openingBracketPos == kNPOS;
       Bool_t defaultDegree = true;
       Int_t degree, counter;
       TString sdegree;
+
+      // Check for extended format supporting direct variable specification
+      Bool_t isNewFormat = false;
       if (!defaultCounter) {
-         // verify first of opening parenthesis belongs to pol expression
-         // character between 'pol' and '(' must all be digits
-         sdegree = formula(polPos + 3, openingBracketPos - polPos - 3);
-         if (!sdegree.IsDigit())
-            defaultCounter = true;
-      }
-      if (!defaultCounter) {
-         degree = sdegree.Atoi();
-         counter = TString(formula(openingBracketPos + 1, formula.Index(')', polPos) - openingBracketPos)).Atoi();
-      } else {
-         Int_t temp = polPos + 3;
-         while (temp < formula.Length() && isdigit(formula[temp])) {
-            defaultDegree = false;
-            temp++;
+         TString args = formula(openingBracketPos + 1, formula.Index(')', polPos) - openingBracketPos - 1);
+         if (args.Contains(",")) {
+            isNewFormat = true;
+            // Extract degree for new format - defaults to 1 if not specified
+            sdegree = formula(polPos + 3, openingBracketPos - polPos - 3);
+            if (!sdegree.IsDigit()) sdegree = "1";
+            degree = sdegree.Atoi();
+            
+            // Handle comma-separated arguments for variable name and parameter start
+            std::unique_ptr<TObjArray> tokens(args.Tokenize(","));
```

Comment:
The modern way would be [std::getline(..., ..., ',')](https://en.cppreference.com/w/cpp/string/basic_string/getline). Could we do this instead of `TObjArray`?